feat: add support for `ES256` signature algorithm

nanderstabel opened the pull request:
# Description of change
Although `did_manager` is capabable of creating and storing a `ES256K` key, we cannot enable it since the library we use for signing (`jsonwebtoken`) we use in `oid4vc` only supports `ES256` and not `ES256K`.

## Links to any relevant issues
- #177 

## How the change has been tested
Manually tested against https://github.com/impierce/ssi-agent/pull/101

In that UniCore is a description of how UniMe can be tested against UniCore.

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

nanderstabel committed:
feat: add support for `ES256`

nanderstabel committed:
fix: fix `client_metadata`

nanderstabel committed:
style: tidy up `Corgo.lock` files

nanderstabel committed:
feat:  add  `pub const SUPPORTED_SIGNING_ALGORITHMS`

nanderstabel committed:
feat: add `IntoJwsAlgorithm` convenience trait

nanderstabel committed:
style: tidy up `stronghold.rs`

nanderstabel committed:
style: remove unused import

nanderstabel committed:
fix: add `client_metadata` to fixture

nanderstabel committed:
style: remove unnecessary map

nanderstabel committed:
style: remove unnecessary references

nanderstabel committed:
fix: retrieve sigining algorithm from `vp_formats` in Authorization Request

nanderstabel committed:
Merge branch 'dev' into feat/es256

nanderstabel committed:
chore: bump did_manager

nanderstabel committed:
Merge branch 'dev' into feat/es256

Norlock commented on Cargo.toml:
```diff
@@ -8,15 +8,16 @@ tauri = { version = "=2.0.0-beta.19", features = [
-log = "^0.4"
-serde_json = "1.0"
-did_manager = { git = "https://github.com/impierce/did-manager.git", rev = "aacecc1" }
+did_manager = { git = "https://github.com/impierce/did-manager.git", rev = "d32ed05" }
```
Maybe add a git tag to the commit and point to that its much clearer to see what we previously pointed to and what commits happened since then. :)

daniel-mader replied:
That's a good idea. I think we should start doing that once we have proper versioning.

Norlock commented on identity-wallet/src/state/credentials/reducers/handle_oid4vp_authorization_request.rs:
```diff
@@ -174,7 +178,7 @@ pub async fn handle_oid4vp_authorization_request(mut state: AppState, action: Ac
 pub fn get_oid4vp_client_name_and_logo_uri(
     oid4vp_authorization_request: &AuthorizationRequest<Object<OID4VP>>,
-) -> (String, Option<String>, String, String) {
+) -> (String, Option<String>, String, String, Algorithm) {
```
Can we create a struct for the return type, it will convey its meaning a bit better

Norlock commented on identity-wallet/src/state/credentials/reducers/handle_oid4vp_authorization_request.rs:
```diff
@@ -183,21 +187,44 @@ pub fn get_oid4vp_client_name_and_logo_uri(
+        ClientMetadataResource::ClientMetadata {
+            client_name,
+            logo_uri,
+            extension: oid4vc::oid4vp::authorization_request::ClientMetadataParameters { vp_formats },
```
maybe remove some of the part and import it e.g.: `authorization_request::ClientMeta...

Norlock commented on identity-wallet/src/state/profile_settings/reducers/create_new.rs:
```diff
@@ -37,14 +38,31 @@ pub async fn create_identity(mut state: AppState, action: Action) -> Result<AppS
-        let did_key = subject.identifier("did:key").await.map_err(|e| Error(e.to_string()))?;
+        let did_key = subject
+            // TODO: make distinction between keys using the same DID Method but different algorithms.
+            .identifier("did:key", Algorithm::EdDSA)
```
Can you make an enum `DidMethod` with variants key, jwk and then add Display trait to it. I think you can pass the enum unless you really need the string in that case you just call .to_string()

daniel-mader committed:
Merge branch 'dev' into feat/es256

daniel-mader committed:
chore: bump deps

daniel-mader committed:
feat: add settings option for preferred key type

daniel-mader committed:
Merge branch 'dev' into feat/es256

daniel-mader committed:
chore: bump tauri deps

daniel-mader committed:
chore: bump oid4vc dep

daniel-mader committed:
feat: add `const` of supported DID methods

daniel-mader committed:
chore: clean up code

daniel-mader committed:
chore(deps): bump `did_manager`

daniel-mader committed:
Merge branch 'dev' into feat/es256

daniel-mader committed:
chore(deps): bump all `tauri` versions

daniel-mader committed:
chore: apply review feedback

daniel-mader committed:
chore: linting issue

daniel-mader committed:
chore: sort ignores, add `coverage/` & untracked vite files

daniel-mader committed:
style: regenerate bindings

nanderstabel committed:
test: update test fixture

daniel-mader committed:
refactor: hide key option `ES256K`

maiertech commented on unime/src/routes/(app)/me/settings/app/did/+page.svelte:
```diff
@@ -9,7 +9,7 @@
   import CheckCircle from '~icons/ph/check-circle-fill';
 
-  $: preferred_did_method = $state.profile_settings.preferred_did_method;
+  $: preferred_did_method = $state.profile_settings.preferred_did_methods.at(0);
```
Why not `preferredDidMethod`? Trying to understand when to use camel case and when snake case.

nanderstabel replied:
From internal discussion (with @daniel-mader):
> We currently export everything without adhering to TypeScript naming conventions. So this is a Rust string, but as a TypeScript value.
> We will change this eventually once we move to a different bindings library, but I'd leave it for now.

maiertech commented on unime/src/routes/(app)/me/settings/app/keys/+page.svelte:
```diff
@@ -0,0 +1,91 @@
+
+  import Info from '~icons/ph/info';
+
+  $: preferred_key_type = $state.profile_settings.preferred_key_types.at(0);
```
Same as before: camel case vs. snake case.

nanderstabel replied:
See above :)

maiertech reviewed (commented):
Fine in principle, but let me have a look again after rebasing.

maiertech reviewed (commented):
Fine in principle, but let me have a look again after rebasing.

nanderstabel committed:
Merge branch 'dev' into feat/es256

nanderstabel committed:
fix: update supported_subject_syntax_types with preferred_did_methods

nanderstabel committed:
Merge branch 'dev' into feat/es256

nanderstabel committed:
fix: use `KeyFillIcon` instead of `Key`

nanderstabel committed:
fix: update the Identity Manager

nanderstabel committed:
chore: execute `npm install` to update the `package-lock.json` file

nanderstabel commented:
I've rebased this branch (and also fixed a bug where the preferred signing algorithm was not being respected), so it is now finally ready for the final review.
